src/app/clusters/descriptor/descriptor.cpp

```diff
@@ -166,7 +174,7 @@ CHIP_ERROR DescriptorAttrAccess::ReadPartsAttribute(EndpointId endpoint, Attribu
 
     case DataModel::EndpointCompositionPattern::kTree:
         return aEncoder.EncodeList([&endpoints, endpoint](const auto & encoder) -> CHIP_ERROR {
-            for (auto & ep : endpoints.GetSpanValidForLifetime())
+            for (auto & ep : endpoints)
```

Comment:
```suggestion
            for (const auto & ep : endpoints)
```

Reply:
Type is already `const ... &` even without this.